src/ophys_etl/transforms/create_slapp_inputs.py

```diff
@@ -0,0 +1,270 @@
+    binarized_basename = argschema.fields.Str(
+        required=True,
+        default="binarize_output.json",
+        description="basename of convert_rois output")
+    trace_basename = argschema.fields.Str(
+        required=True,
+        default="roi_traces.h5",
+        description="basename of convert_rois output")
+
+    @mm.post_load
+    def check_experiment_paths(self, data, **kwargs):
+        missing = False
+        for experiment in data['experiments']:
+            if (('binarized_rois_path' not in experiment) |
+               ('traces_h5_path' not in experiment)):
+                missing = True
+        if missing & ('input_rootdir' not in data):
+            raise ValidationError("either specify 'binarized_rois_path' and "
+                                  "'traces_h5_path' for all experiments or "
+                                  "specify 'input_rootdir'")
+        return data
+
+    @mm.post_load
+    def check_local_ids(self, data, **kwargs):
+        missing = False
+        for experiment in data['experiments']:
+            if ('local_ids' not in experiment):
+                missing = True
+        if missing & ('n_roi_total' not in data):
+            raise ValidationError("either specify 'local_ids' for all "
+                                  "experiments or specify 'n_roi_total' "
+                                  "to trigger random selection")
+        return data
+
+
+class SlappTransformOutputSchema(argschema.schemas.DefaultSchema):
+    outputs = argschema.fields.List(argschema.fields.InputFile)
+
+
+class ExperimentOutputSchema(argschema.schemas.DefaultSchema):
+    experiment_id = argschema.fields.Int(
+        required=True,
+        description="experiment id")
+    local_to_global_roi_id_map = argschema.fields.Dict(
+        keys=argschema.fields.Int(),
+        values=argschema.fields.Int(),
+        required=True,
+        description="{local: global} map")
+    binarized_rois_path = argschema.fields.InputFile(
+        required=True,
+        description="path to output from ophys_etl.convert_rois")
+    traces_h5_path = H5InputFile(
+        required=True,
+        description="path to traces file output from extract_traces")
+    movie_path = H5InputFile(
+        required=True,
+        description="path to mtion corrected movie")
+
+
+def select_rois(experiments: List[Any], n_roi_total: int,
+                random_seed: int) -> List[Any]:
+    """given a list of experiments, subselect a number of non-excluded ROIs
+    and return experiments with selected ROIs with those ids (local) listed in
+    the 'local_ids' field.
+
+    Parameters
+    ----------
+    experiments: list
+        each element satsifies ExperimentSchema
+    n_roi_total: int
+        total number of ROIs to select
+    random_seed: int
+        seed for random number generator
+
+    Returns
+    -------
+    selected_experiments: list
+        each element satsifies ExperimentSchema. 'local_ids' field is populated
+
```

Comment:
Raises
------
ValueError if `n_roi_total` exceeds the total number of ROIs in the `experiments` data

Reply:
ty